File: Fact Digit Sum.java
class Solution
{
    ArrayList<Integer> FactDigit(int N)
    {
        // code here
        ArrayList<Integer> ans=new ArrayList<>();
        int[]fact=new int[10];
        fact[0]=1;
        for(int i=1;i<10;i++){
            fact[i]=i*fact[i-1];
        }
        
        for(int i=9;i>=1;i--){
            while(N>=fact[i]){
                N-=fact[i];
                ans.add(i);
            }
        }
        Collections.reverse(ans);
        return ans;
    }
}

File: Fact Digit Sum.cpp
class Solution{
	public:
	vector<int> FactDigit(int N)
	{
	    // Code here
	    vector<int> fact(10,1);
	    
	    for(int i=1; i<10; i++)
	    fact[i] = (fact[i-1]*i);
	    
	    vector<int> ans;
	    
	    int f = 9;
	    
	    while(N)
	    {
	        if(fact[f] <= N)
	        {
	            N -= fact[f];
	            ans.push_back(f);
	        }
	        else
	        f--;
	    }
	    
	    sort(ans.begin(), ans.end());
	   
	    return ans;
	}
};
